### PR TITLE
Faster svg viewbox computation in tablereport

### PR DESCRIPTION
**Pull request by jeromedockes**
The way the computation of the correct viewbox for the plots is done currently is slow when there are many columns.
This makes all the bounding box computation in one go with no changes to the document in between to avoid unnecessary recomputations of the styles and layout

also as it adds a few functions to the report code, now is a good time to put it in a module to avoid polluting the global namespace so this PR adds that too

### CODE REVIEW

**Review by GaelVaroquaux — APPROVED** [review]
LGTM, but I think that it would be useful to add a short changelog entry: it's user-visible enough

**Comment by jeromedockes** [comment]
I marked it as draft because @Vincent-Maladiere  agreed to check it works in a different environment so we should wait for that info before merging

**Review by Vincent-Maladiere — APPROVED** [review]
LGTM after testing on my machine

**Comment by jeromedockes** [comment]
thanks a lot!

**Pull request merged**
